app/Services/AppointmentService.php

```diff
@@ -118,9 +118,33 @@ public function reject($data, $appointmentId)
 
     public function checkAvailabiltyToReject($appointment)
     {
-        $nowInMs = Carbon::now()->timestamp * 1000;
+        $nowInMs = Carbon::now()->timestamp;
         if ($appointment->status != AppointmentStatus::PENDING || $appointment->time <= $nowInMs) {
             abort(Response::HTTP_BAD_REQUEST,"already rejected or expired");            
         }
     }
+
+    public function getCurrent()
+    {
+        $appointment = $this->repository->getCurrent();
+        if ($appointment->count() == 1) {
+            $appointment = $appointment->first();
+            return $appointment;
+        } else {
+            abort(Response::HTTP_BAD_REQUEST,'not found current appointment to checkin');
+        }
+    }
+    
+    public function checkCurrent($id)
+    {
+        $appointment = $this->show($id);
+        $now = Carbon::now()->timestamp;
+        if ($now < ($appointment->time + $appointment->duration *60) && $appointment->status == AppointmentStatus::APPROVED) {
+            $this->update(['status' => AppointmentStatus::CHECKED] , $appointment->id);
+            return $appointment;
+        } else {
+            abort(Response::HTTP_BAD_REQUEST,'may be appointment is not approved or already completed');
```

Comment:
When writing a code with conditions, write the failed conditions first with its response returning then complete your logic

```
$appointment = $this->show($id);
$now = Carbon::now()->timestamp;

// Check conditions first
if ($now >= ($appointment->time + $appointment->duration *60) || $appointment->status != AppointmentStatus::APPROVED) {
    abort(Response::HTTP_BAD_REQUEST, 'may be appointment is not approved or already completed');
}

$this->update(['status' => AppointmentStatus::CHECKED], $appointment->id);

return $appointment;
```